src/main/java/com/watchtogether/watchtogether/meeting/service/MeetingService.java

```diff
@@ -149,4 +157,83 @@ public MeetingJoinResponseDto joinMeeting(Long meetingId, String memberId) {
         .cinemaName(newMeeting.getCinema().getName())
         .build();
   }
+
+  public List<MeetingHistoryDto> meetinghistory(int page, int size, String memberId) {
+    // 존재하지 않는 사용자라면 예외 밟생
+    Member member = memberRepository.findByMemberId(memberId)
+        .orElseThrow(() -> new MemberNotFoundException("존재하지 않는 사용자입니다."));
+
+    Pageable pageable = PageRequest.of(--page, size); // 페이징을 위한 Pageable 객체 생성
+
+    List<WatchMeetingHistory> list = meetingHistoryService // WatchMeetingHistory -> DTO 변환 필요
+        .readMeetingHistory(memberId, pageable);
+
+    // MeetingHistoryDto List 로 변환 후 return
+    return list.stream().map(data -> MeetingHistoryDto.builder()
+        .canceledDateTime(data.getCancelDateTime())
+        .dateTime(data.getJoinDateTime())
+        .movieTitle(data.getMeeting().getMovie().getTitle())
+        .cinemaName(data.getMeeting().getCinema().getName())
+        .build()).toList();
+  }
+
+  @Transactional
+  public MeetingCancelInfoDto cancelMeeting(String memberId, Long meetingId) {
+    // 존재하지 않는 사용자라면 예외 밟생
+    Member member = memberRepository.findByMemberId(memberId)
+        .orElseThrow(() -> new MemberNotFoundException("존재하지 않는 사용자입니다."));
+
+    // 존재하지 않는 같이볼까요 라면 예외 발생
+    WatchMeeting watchMeeting = meetingRepository.findById(meetingId)
+        .orElseThrow(() -> new MeetingNotFoundException("존재하지 않는 같이볼까요 정보입니다."));
+
+    // 현재날짜 기준 취소하려는 같이볼까요가 3일 이내면 예외 발생
+    if (!checkDate(watchMeeting.getDateTime(), LocalDateTime.now())) {
+      throw new MeetingCancelNotValidDateException("취소 불가능한 날짜의 같이볼까요 입니다.");
+    }
+
+    // 같이볼까요의 현재인원 수정
+    watchMeeting.setNowPeople(watchMeeting.getNowPeople() - 1);
+    WatchMeeting newMeeting = meetingRepository.save(watchMeeting);
+
+    // 포인트 거래내역 추가
+    // 이 usePoint() 내부에 사용자 계정 내부 포인트 변경기능도 포함되어있음.
+    transactionHistoryService.usePoint(member, 15000, TransactionDetail.CANCEL);
+
+    // 신청기록 내부에 취소기록 기입
+    meetingHistoryService.cancelMeetingHistory(member, newMeeting);
+
+    // 만약 모든 인원이 취소했다면 이 같이볼까요는 삭제된다.
+    if (newMeeting.getNowPeople() == 0) {
+      meetingRepository.deleteById(newMeeting.getCode());
+    }
+
+    log.info("{} 님의 같이볼까요 신청이 취소되었습니다. 취소된 같이볼까요 날짜 = {}", memberId, newMeeting.getDateTime());
+
+    return MeetingCancelInfoDto.builder()
+        .movieTitle(newMeeting.getMovie().getTitle())
+        .dateTime(newMeeting.getDateTime())
+        .build();
+  }
+
+  /**
+   * 같이볼까요 취소를 위해 날짜 비교하는 메서드
+   *
+   * @param dateTime 같이볼까요의 날짜
+   * @param now      취소하려는 현재 날짜
+   * @return true == 취소 가능 / false == 취소 불가능
+   */
+  private boolean checkDate(LocalDateTime dateTime, LocalDateTime now) {
+    // A 데이터에서 날짜만 추출
+    LocalDate dateA = dateTime.toLocalDate();
+
+    // B 데이터에서 날짜만 추출
+    LocalDate dateB = now.toLocalDate();
+
+    // 3일 차이 확인
+    if (dateA.minusDays(3).isBefore(dateB)) {
+      return false; // false = 불가능
+    }
+    return true; // 가능
+  }
 }
```

Comment:
Date 관련 함수들은 util성 함수로 빼는 것도 좋을 것 같아요. 하나의 의견으로 봐주시면 좋을 것 같습니다. 
아래와 같은 함수를 별도의 DateUtil 클래스로 빼고, 여기서는 해당 메소드를 가져다 쓰는 형태로 해도 좋을 것 같아요. 
```suggestion
// DateUtil
  public boolean isFewDaysAgo(LocalDateTime dateTime1, LocalDateTime dateTime2, int day) {
    LocalDate date1 = dateTime1.toLocalDate();
    LocalDate date2 = dateTime2.toLocalDate();
    return dateTime1.minusDays(day).isBefore(date2)
}

private boolean checkDate(LocalDateTime dateTime, LocalDateTime now) {
    return !DateUtil.isFewDaysAgo(dateTime, now, 3);
}
```

Reply:
네 멘토님 별도의 DateUtil 클래스 생성하여 Bean 등록 후 사용하는 방법으로 수정했습니다!